### PR TITLE
[12.0] mig partner ref unique

### PR DESCRIPTION
**Pull request by sbejaoui**

### CODE REVIEW

**Review by tbaden — APPROVED** [review]
code review: lgtm

tested in runbot

File: partner_ref_unique/models/res_partner.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+                    (mode == 'companies' and partner.is_company))):
+                domain = [
+                    ('id', '!=', partner.id),
+                    ('ref', '=', partner.ref),
+                ]
+                if mode == 'companies':
+                    domain.append(('is_company', '=', True))
+                other = self.search(domain)
+
+                # active_test is False when called from
```
Comment (NL66278) [inline partner_ref_unique/models/res_partner.py:27]: Might there not be a lot of other situations where active_test is False? In all these cases the constraint will be disabled...

Comment (tbaden) [reply partner_ref_unique/models/res_partner.py:27]: as far as I can see the odoo core will only set active_test to false in base.partner.merge.automatic.wizard, like the comment stated. only 3rd party modules could bypass this.

**Review by NL66278 — APPROVED** [review]
:+1: Remark on ActiveTest not blocking, for the rest LGTM.

**Pull request merged**